lib/shared/src/chat/prompts/vscode-context.ts

```diff
@@ -104,65 +121,136 @@ export async function getCurrentDirContext(isTestOnly: boolean): Promise<Context
     return getEditorDirContext(currentDir, currentFile, isTestOnly)
 }
 
-/**
- * Gets editor directory context.
- *
- * @param directoryPath - The path to the directory to get context for.
- * @param currentFileName - Optional current file name.
- * @param onlyTests - Whether to only include test files in the context.
- * Default is false.
- *
- * @returns A promise that resolves to the context messages for the directory.
- */
 export async function getEditorDirContext(
```

Comment:
Some docs to explain this would be cool. I recommend
```suggestion
/**
 * Asynchronously gets the context messages from a specified directory.
 *
 * @param directoryPath - The absolute path to the target directory.
 * @param currentFileName - The name of the current file. Optional.
 * @param testFilesOnly - If set to true, function will only consider test files. Default value is false.
 *
 * If `testFilesOnly` is true and `currentFileName` is provided, function first attempts to get the context from filtered
 * files in the current directory, if any exist. If there's no context found, it seeks context from the editor test. If no context
 * is found in both cases, it defaults to getting context from the first 10 files in the directory.
 *
 * If `testFilesOnly` is false, it directly gets the context from the first 10 files in the directory.
 *
 * @returns - A promise that resolves with the array of context messages.
 */
export async function getEditorDirContext(
```